go-controller/pkg/node/gateway_iptables.go

```diff
@@ -77,9 +78,13 @@ func addIptRules(rules []iptRule) error {
 			klog.V(5).Infof("Chain: \"%s\" in table: \"%s\" already exists, skipping creation: %v",
 				r.chain, r.table, err)
 		}
-		exists, err := ipt.Exists(r.table, r.chain, r.args...)
+		exists, err = ipt.Exists(r.table, r.chain, r.args...)
 		if !exists && err == nil {
-			err = ipt.Insert(r.table, r.chain, 1, r.args...)
+			if append {
+				err = ipt.Append(r.table, r.chain, r.args...)
+			} else {
+				err = ipt.Insert(r.table, r.chain, 1, r.args...)
+			}
 		}
 		if err != nil {
 			addErrors = errors.Wrapf(addErrors, "failed to add iptables %s/%s rule %q: %v",
```

Comment:
Maybe this is out of the PR scope, but `addErrors` is always `nil`, as it is its starting value:

https://go.dev/play/p/gMyA00mvOXB

Reply:
yeah nice catch will do!

Reply:
done.